modules/sfm/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+set(the_description "SFM algorithms")
+
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Winvalid-pch)
+
+# needed since we are linking static libs
+set(CMAKE_CXX_FLAGS ${CMAKE_CSS_FLAGS} "-fPIC")
```

Comment:
This flag doesn't work with MSVS.
Please use something like this:
https://github.com/Itseez/opencv/blob/09b9b0fb9e9c9dd8c9e0d65705f8f19aa4c27f8a/3rdparty/zlib/CMakeLists.txt#L79-L83